### PR TITLE
memory leak fixes and fetchable sig enablement

### PR DESCRIPTION
**Pull request by baentsch**
This corrects some memory leaks detected by `valgrind` and adds logic to support #9. 

Until https://github.com/openssl/openssl/issues/10512 is fully resolved upstream, the new logic can be exercized by running the openssl version in https://github.com/baentsch/openssl/tree/sigload.

### CODE REVIEW

File: scripts/runtests.sh [hunk 1]
```diff
@@ -119,8 +116,11 @@ interop p256_sphincsshake256128frobust
 interop rsa3072_sphincsshake256128frobust
 ##### OQS_TEMPLATE_FRAGMENT_ALGS_END
 
+# Run built-in tests:
+(cd _build; ctest $@)
+
 # cleanup
-rm -rf tmp
+#rm -rf tmp
```
Comment (bhess) [inline scripts/runtests.sh:123]: No cleanup needed here?

File: test/oqs_test_tlssig.c [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    if (!strcmp(provname, PROVIDER_NAME_OQS))
+        return OSSL_PROVIDER_get_capabilities(provider, "TLS-SIGALG",
+                                              test_signature, vctx);
+    else
+        return 1;
+}
+
+#define nelem(a) (sizeof(a)/sizeof((a)[0]))
+
+static size_t tcb(const char *buf, size_t cnt,
```
Comment (bhess) [inline test/oqs_test_tlssig.c:129]: Function seems to be unused with the commented-out section in main(). Maybe add a ifdef?

File: test/oqs_test_tlssig.c [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        break;
+    case OSSL_TRACE_CTRL_END:
+        label = "END";
+        break;
+    }
+
+    if (label != NULL) {
+        union {
+            pthread_t tid;
+            unsigned long ltid;
```
Comment (bhess) [inline test/oqs_test_tlssig.c:147]: Is this portable? Maybe cast pthread_t instead.

**Review by bhess — COMMENTED** [review]
Does it mean that the code is not compatible with upstream openssl at the moment (circleci seems to get openssl from upstream)?

Looks good to me, only minor points below.

File: scripts/runtests.sh (hunk 1 above)
Comment (baentsch) [reply scripts/runtests.sh:125]: Not yet. See added "TBC" comment.

File: test/oqs_test_tlssig.c (hunk 3 above)
Comment (baentsch) [reply test/oqs_test_tlssig.c]: removed.

**Comment by baentsch** [comment]
ping @bhess

**Review by bhess — APPROVED** [review]

**Pull request merged**